### PR TITLE
Get Consumer port from listener if hostname is configured

### PR DESCRIPTION
**Pull request by mageshn**

### CODE REVIEW

**Comment by mageshn** [comment]
@ewencp when you get a chance, can you take a look

**Comment by ewencp** [comment]
@mageshn could someone else on @confluentinc/clients review this? is there a reason i should be digging in?

File: src/main/java/io/confluent/kafkarest/KafkaRestConfig.java [hunk 1]
```diff
@@ -740,6 +757,8 @@ private  String getBootstrapBrokers(ZkUtils zkUtils) {
     return bootstrapBrokers;
   }
 
+
+
```
Comment (edenhill) [inline src/main/java/io/confluent/kafkarest/KafkaRestConfig.java]: nit: whitespace diff

File: src/test/java/io/confluent/kafkarest/unit/UriUtilsTest.java [hunk 2]
```diff
@@ -79,4 +79,34 @@ public void testAbsoluteURIBuilderWithPort() throws RestConfigException {
     EasyMock.verify(uriInfo);
   }
 
+  @Test
+  public void testAbsoluteURIBuilderWithListenerForHttp() throws RestConfigException {
+    Properties props = new Properties();
+    props.put(KafkaRestConfig.HOST_NAME_CONFIG, "bar.net");
+    props.put(KafkaRestConfig.LISTENERS_CONFIG, "http://0.0.0.0:9091,https://0.0.0.0:9092");
```
Comment (edenhill) [inline src/test/java/io/confluent/kafkarest/unit/UriUtilsTest.java]: add tests for IPv6 addresses:
 * full address, no port - e.g. `fe80:0:1:2:3:4:5:6`
 * full address with port - e.g., `[fe80:0:1:2:3:4:5:6]:9092` 
 * truncated address, no port, e.g., `fe80::1`
 * truncated address, with port `[fe80::1]:9092`

**Comment by mente** [comment]
out of curiosity - since what stable version this PR is included? https://github.com/confluentinc/kafka-rest/blob/master/docs/changelog.rst doesn't have it

**Comment by ewencp** [comment]
@mente Since 4.1.0, not sure how it slipped by the changelog

**Pull request merged**